### PR TITLE
fix(grpc): remove package from type names

### PR DESCRIPTION
**Pull request by shashitnak**
**Build & Testing:**

- [x] I ran `cargo test` successfully.
- [x] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [ ] I have added relevant unit & integration tests.
- [ ] I have updated the [documentation] accordingly.
- [ ] I have performed a self-review of my code.
- [ ] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

## Summary by CodeRabbit

- **Refactor**
	- Enhanced the handling of package data structures to improve performance and flexibility.

### CODE REVIEW

File: src/generator/snapshots/tailcall__generator__from_proto__test__from_proto.snap [hunk 1]
```diff
@@ -36,6 +36,11 @@ input NEWS_NEWS_ID @tag(id: "news.NewsId") {
   id: Int
 }
 
+enum NEWS_STATUS @tag(id: "Status") {
```
Comment (tusharmath) [inline src/generator/snapshots/tailcall__generator__from_proto__test__from_proto.snap:39]:
```suggestion
enum NEWS_STATUS @tag(id: "news.Status") {
```

Package is missing in the id.

File: src/generator/graphql_type.rs [hunk 2]
```diff
@@ -121,6 +140,14 @@ impl GraphQLType<Parsed> {
...
         }
     }
+
+    pub fn name(&self) -> String {
+        self.0.name.clone()
+    }
+
+    pub fn package(&self) -> Option<String> {
+        self.0.package.as_ref().map(|x| x.source().to_string())
+    }
```
Comment (tusharmath) [inline src/generator/graphql_type.rs:150]: we can revert this.

File: src/generator/graphql_type.rs [hunk 3]
```diff
@@ -121,6 +139,10 @@ impl GraphQLType<Parsed> {
             self.0.name.clone()
         }
     }
+
+    pub fn name(&self) -> String {
+        self.0.name.clone()
+    }
```
Comment (tusharmath) [inline src/generator/graphql_type.rs:145]: drop this

File: src/generator/snapshots/tailcall__generator__from_proto__test__from_proto.snap [hunk 4]
```diff
@@ -36,6 +36,11 @@ input NEWS_NEWS_ID @tag(id: "news.NewsId") {
   id: Int
 }
 
+enum NEWS_STATUS @tag(id: "news.Status") {
+  DRAFT
+  PUBLISHED
+}
```
Comment (tusharmath) [inline src/generator/snapshots/tailcall__generator__from_proto__test__from_proto.snap:42]: Add tests for messages inside of messages.

**Review by tusharmath — APPROVED** [review]

File: src/generator/from_proto.rs [hunk 5]
```diff
@@ -59,7 +59,11 @@ impl Context {
 
             ty.variants = Some(variants);
 
-            let type_name = GraphQLType::new(enum_name).as_enum().unwrap().to_string();
+            let type_name = GraphQLType::new(enum_name)
+                .package(&self.package)
+                .as_enum()
+                .unwrap()
+                .to_string();
```
[the remark this answers is not in the document]
Comment (meskill) [reply src/generator/from_proto.rs]: actually, the `GraphQLType<Unparsed>::parse` that is used by `as_enum` could be rewritten without return Option that could simplify usage of it and remove unwraps

File: src/generator/snapshots/tailcall__generator__from_proto__test__from_proto.snap (hunk 4 above)
Comment (meskill) [reply src/generator/snapshots/tailcall__generator__from_proto__test__from_proto.snap:42]: news.proto already contains message inside of message https://github.com/tailcallhq/tailcall/pull/1747/files#diff-771adfaaae91c7816aaf69af02c57749bfaaaa74e3db399e055104a8a1a6de23R26-R33

is it enough?

**Pull request closed without merging**